File: 781-rabbits-in-forest/781-rabbits-in-forest.cpp
class Solution {
public:
    int numRabbits(vector<int>& answers) {
        vector<int>cnt(1001,0);
        for(auto ans:answers){
            cnt[ans]++;
        }
        int total = 0;
        for(int i=0;i<1001;i++){
            if(cnt[i]){
                int a = i+1;
                int d = cnt[i]/a;
                if(cnt[i]%a) d++;
                total+=(d*a);
            }
        }
        return total;
    }
};

File: 781-rabbits-in-forest/781-rabbits-in-forest.py
class Solution:
    def numRabbits(self, answers: List[int]) -> int:
        cnt = [0 for _ in range(1001)]
        for ans in answers:
            cnt[ans]+=1
        total = 0
        for i in range(0,1001):
            if(cnt[i]>0):
                a = i+1;
                d = cnt[i]//a;
                if(cnt[i]%a):
                    d+=1;
                total+=d*a;
        return total